k2/python/k2/fsa_algo.py

```diff
@@ -257,3 +258,58 @@ def add_epsilon_self_loops(fsa: Fsa) -> Fsa:
         setattr(out_fsa, name, value)
 
     return out_fsa
+
+
+def remove_epsilon(fsa: Fsa) -> Fsa:
+    '''Remove epsilons (symbol zero) in the input Fsa.
+
+    Caution:
+      It only works on for CPU and doesn't support autograd.
+
+    Args:
+      fsa:
+        The input FSA. It can be either a single FSA or a FsaVec.
+        Must be top-sorted.
+    Returns:
+        The result Fsa, it's equivalent to the input ``fsa`` under
+        tropical semiring but will be epsilon-free.
+        It will be the same as the input ``fsa`` if the input
+        ``fsa`` is epsilon-free. Otherwise, a new epsilon-free fsa
+        is returned and the input ``fsa`` is NOT modified.
+    '''
+    properties = getattr(fsa, 'properties', None)
+    if properties is not None and is_epsilon_free(properties):
+        return fsa
+
+    ragged_arc = _k2.remove_epsilon(fsa.arcs)
+    out_fsa = Fsa.from_ragged_arc(ragged_arc)
```

Comment:
The non-tensor attributes can be copied without an arc map.

Reply:
Just a question, @csukuangfj , How do we copy aux_labels? it's a non-tensor or tensor, I was thinking it's a tensor attribute, but didn't find code in fsa.py which shows it's a tensor attribute

Reply:
https://github.com/k2-fsa/k2/blob/6013e15598ed673d4eec34dcf2f59b8bc47717a7/k2/python/k2/fsa.py#L113-L114

`self.aux_labels` will invoke 
https://github.com/k2-fsa/k2/blob/6013e15598ed673d4eec34dcf2f59b8bc47717a7/k2/python/k2/fsa.py#L206

https://github.com/k2-fsa/k2/blob/6013e15598ed673d4eec34dcf2f59b8bc47717a7/k2/python/k2/fsa.py#L215-L222

`aux_labels` is just like other attributes. If it is of type `torch.Tensor`, it is saved in `self._tensor_attr`.

Reply:
Ha, many thanks!